nodes/clarify_util.js

```diff
@@ -195,6 +207,48 @@ function validateMapStringWithStrings(validationErrors, varName, variable) {
   return validationErrors;
 }
 
+function validateMapStringWithString(validationErrors, varName, variable) {
+  if (variable === undefined) {
+    return;
+  }
+
+  if (typeof variable !== 'object') {
+    validationErrors.push(varName + ' must be object');
+    return;
+  }
+
+  for (const [key, values] of Object.entries(variable)) {
+    if (typeof key !== 'string') {
+      validationErrors.push(key + ' in ' + varName + ' must be a string');
+      continue;
+    }
+
+    if (key === '') {
+      validationErrors.push(varName + ' keys can not be empty');
+      continue;
+    }
+
+    if (key.length > 40) {
+      validationErrors.push(key + ' in ' + varName + ' is too long. Max 40 chars.');
+    }
+
+    if (key.indexOf(' ') >= 0) {
+      validationErrors.push(key + ' in ' + varName + ' can not contain spaces.');
+    }
+
+    if (!keyPattern.test(key)) {
+      validationErrors.push(key + ' in ' + varName + ' must fulfil ' + keyPattern);
+    }
```

Comment:
Should not the last line be enough here?

```
  if (!keyPattern.test(key)) {
      validationErrors.push(key + ' in ' + varName + ' must fulfil ' + keyPattern);
    }
```

If the regex is set up correctly, it will match  all of the other cases mentioned above and be easier to maintain.

Same goes for the `validateMapStringWithStrings` method.

Reply:
You're correct. I added the two checks to generate more specific error messages, but the last one covers it all. I'll remove the first two checks for both functions.